source/service.go

```diff
@@ -121,20 +122,38 @@ func (sc *serviceSource) endpointsFromTemplate(svc *v1.Service) ([]*endpoint.End
 		return nil, fmt.Errorf("failed to apply template on service %s: %v", svc.String(), err)
 	}
 
+	ttl := getTTLFromAnnotations(svc.Annotations)
 	hostname := buf.String()
 	for _, lb := range svc.Status.LoadBalancer.Ingress {
 		if lb.IP != "" {
 			//TODO(ideahitme): consider retrieving record type from resource annotation instead of empty
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.IP, ""))
+			endpoints = append(endpoints, endpoint.NewEndpointWithTTL(hostname, lb.IP, "", ttl))
 		}
 		if lb.Hostname != "" {
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.Hostname, ""))
+			endpoints = append(endpoints, endpoint.NewEndpointWithTTL(hostname, lb.Hostname, "", ttl))
 		}
 	}
 
 	return endpoints, nil
 }
 
+func getTTLFromAnnotations(annotations map[string]string) endpoint.TTL {
+	ttlAnnotation, exists := annotations[ttlAnnotationKey]
+	if !exists {
+		return endpoint.TTL{IsConfigured: false}
+	}
+	ttlValue, err := strconv.ParseInt(ttlAnnotation, 10, 64)
```

Comment:
`strconv.Atoi` returns `int` whereas `ttl` is `int64`. I could convert but I thought this is the most direct route to get what I need.